### DISCUSSION TITLE
Temporarily disable extras matching?

### DISCUSSION THREAD

**Discussion by kopecs** [tree-sitter/tree-sitter#1771] [Q&A] [answered]
I'm having some trouble with a situation where I'd like to be able to disable extras matching before an external rule (<https://github.com/tree-sitter/tree-sitter-html/issues/40>---inside of an HTML script tag I'd like to disable extras so that an external scanner can consume all of the whitespace, rather than some be lost to extras).

I've taken a look at https://github.com/tree-sitter/tree-sitter/issues/861 and https://github.com/tree-sitter/tree-sitter/issues/1087 but I haven't been able to get a working solution. Simply lowering the precedence of whitespace in extras doesn't seem to work, which is what #1087 seems to suggest, though I might be misunderstanding it. Additionally, https://github.com/tree-sitter/tree-sitter/issues/931 seems related but doesn't seem to have a resolution.

A (perhaps over)simplified example of my case is:
```
extras: $ => [ /\s+/ ],
```
with a rule like
```
    script_element: $ => seq(
      token("<script>"),
      optional($.raw_text),
      token("</script>")
    ),
```
where `raw_text` is external and should match all of the contained text, leading whitespace included.

For a completely accurate case see https://github.com/tree-sitter/tree-sitter-html/blob/29f53d8f4f2335e61bf6418ab8958dac3282077a/grammar.js#L51

**Comment by ahelwer** [comment, marked as answer]
This is kind of mysterious, because in my experience the external scanner will always be called as soon as one of the external tokens could be valid at that position. It takes precedence over the extras. Are you sure your external scanner tokens are valid when you think they are?

**Reply by kopecs** [reply]
I had a chance to look at this more today and had missed an earlier advance in a calling function.

Thanks!
